### ISSUE TITLE
CVE-2023-5129

### ISSUE THREAD

**Issue by lje** [kiwibrowser/src.next#1001]
**Describe the bug**
Kiwi needs to fix CVE-2023-5129 ASAP

I'm pretty sure there was an open issue for CVE-2023-5129, but somehow it disappeared.

**Comment by htsmi** [comment]
Also CVE-2023-5217. It doesn't look so great when serious security issues mysteriously disappear from the issue tracker. I'll just hope it was an accident, because I wouldn't want to consider the implications of the alternative.

**Comment by ggtylerr** [comment]
Glad I'm not the only one that noticed it. Here's the original email notifications as proof:
![Screenshot of two emails for a deleted issue, #989, regarding CVE-2023-4863](https://i.ggt.bz/ivr8au7cfves.jpg)

**Comment by 0ip** [comment]
How certain are we that Kiwi is indeed affected by this issue, i.e. uses its own (old) libwebp library?
Is there a website that allows you to check if a browser is vulnerable?

**Comment by htsmi** [comment]
> How certain are we that Kiwi is indeed affected by this issue, i.e. uses its own (old) libwebp library? Is there a website that allows you to check if a browser is vulnerable?

Not certain. But it seems likely given that the browser renders webp and vpx files and hasn't been given the latest patches. Enough that I'm not using it in the meantime.

**Comment by ggtylerr** [comment]
It's also worth mentioning that the CVEs were published in late September, while the last commit was in mid August.

**Comment by 0ip** [comment]
> [The Chromium version](https://github.com/kiwibrowser/src.next/blob/64f705f0f92204892c61ac769d12c21a60e327cf/CHROMIUM_VERSION) in Kiwi's latest commit says it's in `105.0.5195.24`.

@ggtylerr Interesting, because I took a look at what Kiwi says on my Android phone (installed via the Playstore) and it reports `116.0.5845.72`. Therefore, much more recent than 105, but still affected, apparently.

@htsmi Fair enough! I thought of the possibility that Kiwi may render webp (and/or even vpx) media using a system wide API or library. But I have to admit that I have no clear understanding of the internals. I just wanted to bring the possibility into play.

**Comment by ggtylerr** [comment]
> @ggtylerr Interesting, because I took a look at what Kiwi says on my Android phone (installed via the Playstore) and it reports `116.0.5845.72`. Therefore, much more recent than 105, but still affected, apparently.

That is a bit interesting but it turns out that [there's an action workflow that automatically updates chromium and makes a release.](https://github.com/kiwibrowser/src.next/blob/kiwi/.github/workflows/rebase_kiwi_on_top_of_chromium.yml) Here's [the release that has that version.](https://github.com/kiwibrowser/src.next/releases/tag/5877778038)

Either way, the release was made a full month before the two vulns were published, so it's definitely not patched.

**Comment by lje** [comment]
@kiwibrowser do you intend to release an updated version soon? If you're not able to fix security holes in a timely manner, it is not responsible to continue the project (IMO).

**Comment by ggtylerr** [comment]
> @kiwibrowser do you intend to release an updated version soon? If you're not able to fix security holes in a timely manner, it is not responsible to continue the project (IMO).

To be fair, it is a small development team doing FOSS work. They have no obligations and the browser itself is pretty damn good.

That being said, it's very troubling that they haven't even commented on this, much less release a patch. The only response so far is the issue being deleted, which, like @htsmi said, doesn't look good. 

I'll be notifying the FFUpdater team and Google soon if this isn't resolved. These are two major vulns and they're still publicly available without any notice.

**Comment by lje** [comment]
> To be fair, it is a small development team doing FOSS work. They have no obligations and the browser itself is pretty damn good.

That's right. It is still irresponsible to keep a software with known serious vulnerabilities ready for download. It should at least be removed from Google Play.

**Comment by 0ip** [comment]
New release 🎉
https://github.com/kiwibrowser/src.next/releases/tag/6453943399
now 116.0.5845.240
